Return-Path: <olaf@suse.de>
Received: from hermes.suse.de (Hermes.suse.de [10.0.0.1])
	by wotan.suse.de (Postfix) with ESMTP id A8FF510616
	for <olh@wotan.suse.de>; Tue, 17 Feb 2004 06:14:06 +0100 (CET)
Received: by hermes.suse.de (Postfix)
	id 9F1738988; Tue, 17 Feb 2004 06:14:06 +0100 (CET)
Received: from mandarine.suse.de (Mandarine.suse.de [10.10.3.21])
	by hermes.suse.de (Postfix) with ESMTP
	id 981783657; Tue, 17 Feb 2004 06:14:06 +0100 (CET)
Received: by mandarine.suse.de (Postfix, from userid 90)
	id 81F3246A17; Tue, 17 Feb 2004 06:14:06 +0100 (CET)
To: mantel@suse.de, olh@suse.de
Subject: Incoming changes to linux-2.5 cset 1.1712  mandarine.suse.de:/olaf/sources/tree/linux-2.5 
Message-Id: <20040217051406.81F3246A17@mandarine.suse.de>
Date: Tue, 17 Feb 2004 06:14:06 +0100 (CET)
From: olaf@suse.de (Olaf Hering)
X-Spam-Checker-Version: SpamAssassin 2.63 (2004-01-11) on wotan.suse.de
X-Spam-Level: 
X-Spam-Status: No, hits=-4.8 required=5.0 tests=AWL,BAYES_00 autolearn=ham 
	version=2.63
Content-Length: 2612

old status OK
ChangeSet
  1.1712 04/02/16 16:20:24 torvalds@home.osdl.org +2 -0
  Revert the dodgy ia64 serial console changeset by Bjorn Helgaas.
  
  It results in serial console getting initialised really late and the
  suggested workaround is broken according to Keith. 
  
  Cset exclude: akpm@osdl.org|ChangeSet|20040213234712|28554

  drivers/serial/serial_core.c
    1.80 04/02/16 16:20:22 torvalds@home.osdl.org +0 -0
    Exclude

  drivers/serial/8250.c
    1.45 04/02/16 16:20:22 torvalds@home.osdl.org +0 -0
    Exclude

.........................................................................
# This is a BitKeeper generated patch for the following project:
# Project Name: Linux kernel tree
# This patch format is intended for GNU patch command version 2.5 or higher.
# This patch includes the following deltas:
#	           ChangeSet	1.1711  -> 1.1712 
#	drivers/serial/8250.c	1.44    -> 1.45   
#	drivers/serial/serial_core.c	1.79    -> 1.80   
#
# The following is the BitKeeper ChangeSet Log
# --------------------------------------------
# 04/02/16	torvalds@home.osdl.org	1.1712
# Revert the dodgy ia64 serial console changeset by Bjorn Helgaas.
# 
# It results in serial console getting initialised really late and the
# suggested workaround is broken according to Keith. 
# 
# Cset exclude: akpm@osdl.org|ChangeSet|20040213234712|28554
# --------------------------------------------
#
diff -Nru a/drivers/serial/8250.c b/drivers/serial/8250.c
--- a/drivers/serial/8250.c	Tue Feb 17 06:14:05 2004
+++ b/drivers/serial/8250.c	Tue Feb 17 06:14:05 2004
@@ -1976,8 +1976,6 @@
 	if (co->index >= UART_NR)
 		co->index = 0;
 	port = &serial8250_ports[co->index].port;
-	if (port->type == PORT_UNKNOWN)
-		return -ENODEV;
 
 	/*
 	 * Temporary fix.
@@ -2008,14 +2006,6 @@
 	return 0;
 }
 console_initcall(serial8250_console_init);
-
-static int __init serial8250_late_console_init(void)
-{
-	if (!(serial8250_console.flags & CON_ENABLED))
-		register_console(&serial8250_console);
-	return 0;
-}
-late_initcall(serial8250_late_console_init);
 
 #define SERIAL8250_CONSOLE	&serial8250_console
 #else
diff -Nru a/drivers/serial/serial_core.c b/drivers/serial/serial_core.c
--- a/drivers/serial/serial_core.c	Tue Feb 17 06:14:05 2004
+++ b/drivers/serial/serial_core.c	Tue Feb 17 06:14:05 2004
@@ -1871,6 +1871,9 @@
 	if (flow == 'r')
 		termios.c_cflag |= CRTSCTS;
 
+	if (!port->ops)
+		return 0;	/* "console=" on ia64 */
+
 	port->ops->set_termios(port, &termios, NULL);
 	co->cflag = termios.c_cflag;
 
.........................................................................
# vim: syntax=diff

